server/src/main/java/org/opensearch/extensions/action/ExtensionActionResponse.java

```diff
@@ -41,7 +48,16 @@ public ExtensionActionResponse(byte[] responseBytes) {
      * @throws IOException when message de-serialization fails.
      */
     public ExtensionActionResponse(StreamInput in) throws IOException {
-        responseBytes = in.readByteArray();
+        this.success = in.readBoolean();
```

Comment:
So the ExtensionActionResponse is already released (in 2.6 at least): https://github.com/opensearch-project/OpenSearch/blob/2.6/server/src/main/java/org/opensearch/extensions/action/ExtensionActionResponse.java


```suggestion
if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
        this.success = in.readBoolean();
}
```
```

Reply:
Yes but it's marked `@opensearch.internal` and is only used by Extensions (not yet released) and will be needed for Extensions GA sometime in 2.X.

Reply:
That's fine for `@opensearch.internal` but bwc tests will fail in mixed clusters, I suspect we don't have any tests for extensions yet